### PR TITLE
chore: optimize metrics creation and add documentation

### PR DESCRIPTION
**Pull request by azrod**

### CODE REVIEW

**Comment by azrod** [comment]
Ignore commit pre-commit ci

File: internal/metrics/admission-controller.go [hunk 1]
```diff
@@ -5,48 +5,24 @@ import (
 )
 
 type (
-	admissionController struct{}
+	admissionController struct {
+		Total         prometheus.Counter `help:"The total number of action performed."`
```
Comment (dmicheneau) [inline internal/metrics/admission-controller.go:9]: is not an 'action' is a number of time 'admission controller' is call

**Review by dmicheneau — CHANGES_REQUESTED** [review]

File: cmd/admission-controller/webhook.go [hunk 2]
```diff
@@ -139,16 +139,16 @@ func mutate(ctx context.Context, ar *admissionv1.AdmissionReview) *admissionv1.A
...
+	timePatch := metrics.AdmissionController().PatchDuration.NewTimer()
 	defer timePatch.ObserveDuration()
 
 	var err error
 	// find annotation enabled
 	an := annotations.New(ctx, pod)
 	if !an.Enabled().Get() {
 		// increment the total number of errors
-		metrics.AdmissionControllerPatch().TotalErr().Inc()
+		metrics.AdmissionController().PatchErrTotal.Inc()
```
Comment (dmicheneau) [inline cmd/admission-controller/webhook.go:151]: change it by PatchErrorTotal

File: internal/metrics/admission-controller.go [hunk 3]
```diff
@@ -5,48 +5,24 @@ import (
 )
 
 type (
-	admissionController struct{}
+	admissionController struct {
+		RequestTotal      prometheus.Counter `help:"The total number of request received."`
+		RequestErrorTotal prometheus.Counter `help:"The total number of request received with error."`
+		RequestDuration   Histogram          `help:"The duration in seconds of request in admission controller."`
+		PatchTotal        prometheus.Counter `help:"The total number of patch action performed."`
+		PatchErrTotal     prometheus.Counter `help:"The total number of patch action performed with error."`
```
Comment (dmicheneau) [inline internal/metrics/admission-controller.go:13]: Suggest to uniform: PatchErrorTotal

File: internal/metrics/events.go [hunk 4]
```diff
@@ -5,49 +5,21 @@ import (
...
-// The counter is used to observe the number of events that failed.
-// The counter is incremented each time an event fails.
-// A good practice is to use the following pattern:
-//
-// metrics.Events().TotalErr().Inc()
-func (a *events) TotalErr() prometheus.Counter {
-	return eventsTotalErr
-}
+func Events() events {
+	if eventsMetrics.TriggerdTotal == nil {
```
Comment (dmicheneau) [inline internal/metrics/events.go:20]:
```suggestion
	if eventsMetrics.TriggeredTotal == nil {
```

File: internal/metrics/events.go [hunk 5]
```diff
@@ -5,49 +5,21 @@ import (
 )
 
 type (
-	events struct{}
+	events struct {
+		TriggerdTotal      prometheus.Counter `help:"The total number of events triggered."`
```
Comment (dmicheneau) [inline internal/metrics/events.go:9]:
```suggestion
		TriggeredTotal      prometheus.Counter `help:"The total number of events triggered."`
```

File: test/metrics/metrics_test.go [hunk 6]
```diff
@@ -71,10 +75,8 @@ func TestMetric_Counter(t *testing.T) {
 		// Test the metrics for the actions Counter
 		for _, tt := range testUnit {
 			t.Run(tt.name, func(t *testing.T) {
-				counter := tt.c
-				counter.Inc()
 				// Compare the metrics
-				if err := testutil.CollectAndCompare(counter, strings.NewReader(tt.data+tt.nameMetric+tt.value), tt.nameMetric); err != nil {
+				if err := testutil.CollectAndCompare(tt.c, strings.NewReader(tt.data+tt.nameMetric+tt.value), tt.nameMetric); err != nil {
```
Comment (dmicheneau) [inline test/metrics/metrics_test.go:79]:
```suggestion
				if err := testutil.CollectAndCompare(counter, strings.NewReader(tt.data+tt.nameMetric+tt.value), tt.nameMetric); err != nil {
```
In logic, we need to compare the object that we have increment before.

**Review by dmicheneau — CHANGES_REQUESTED** [review]

**Review by dmicheneau — APPROVED** [review]

**Pull request merged**